src/main/java/com/pmm/budgetcalculator/expensing/service/ExpenseTypeService.java

```diff
@@ -23,18 +24,32 @@ public ExpenseType getExpenseTypeById(Long id) {
                 .findById(id)
                 .orElseThrow(() -> new RuntimeException("Expense not found for Id:" + id));
     }
+
     public ExpenseType createExpenseType(ExpenseType expenseType) {
         return expenseTypeRepository.save(expenseType);
     }
-    public void removeExpenseTypeById(Long id){
+
+    public void removeExpenseTypeById(Long id) {
         if (expenseTypeRepository.existsById(id)) {
             expenseTypeRepository.deleteById(id);
         }
+    }
 
+    public ResponseEntity<ExpenseType> updateExpenseType(Long id, ExpenseType expenseType) {
+        if (expenseTypeRepository.existsById(id))
+        {
+            final ExpenseType updateExpenseType = expenseTypeRepository.save(expenseType);
+            return ResponseEntity.ok(updateExpenseType);
+        }
+        throw new RuntimeException("User not found on :: "+ id);
```

Comment:
Runtime Exception is too general to be thrown anywhere :). Also message is miss leading.